tests/zfs-tests/tests/functional/cli_user/zpool_iostat/zpool_iostat_005_pos.ksh

```diff
@@ -54,10 +54,27 @@ fi
 # a '/' when we specify the path (-P) flag. We check for "{}" to see if one
 # of the VDEV variables isn't set.
 #
-C1=$($ZPOOL iostat -Pv | $GREP -E '^\s+/' | $WC -l)
-C2=$($ZPOOL iostat -Pv -c 'echo vdev_test{$VDEV_PATH}{$VDEV_UPATH}' | $GREP -E '^\s+/' | $GREP -v '{}' | $WC -l)
+C1=$($ZPOOL iostat -Pv $testpool | $GREP -E '^\s+/' | $WC -l)
+C2=$($ZPOOL iostat -Pv -c 'echo vdev_test{$VDEV_PATH}{$VDEV_UPATH}' $testpool \
+    | $GREP -E '^\s+/' | $GREP -v '{}' | $WC -l)
 if [ "$C1" != "$C2" ] ; then
 	log_fail "zpool iostat -c failed, expected $C1 vdevs, got $C2"
 else
-	log_pass "zpool iostat -c passed, expected $C1 vdevs, got $C2"
+	log_note "zpool iostat -c passed, expected $C1 vdevs, got $C2"
+fi
+
+# Call iostat on only a specific vdev, and verify that the command only gets
+# run on the vdev.  We write the command results to a temp file to verify that
+# the command actually gets run, rather than just verifying that the results
+# are *displayed* for the specific vdev.
+TMP=$($MKTMP)
+FIRST_VDEV=$($ZPOOL iostat -Pv $testpool | $GREP -Eo '^\s+/[^ ]+' | $HEAD -n 1)
+log_must $ZPOOL iostat -Pv -c "'echo \$VDEV_PATH >> "$TMP"'" $testpool \
+    $FIRST_VDEV > /dev/null
+C2=$($WC -l <<< $TMP)
```

Comment:
Is `<<<` a ksh thing?  I think you mean `C2=$($WC -l $TMP | $AWK 'print $1 }')` here to get the line count.  This happened to work because it would always result in 1.